src/box/lua/net_box.lua

```diff
@@ -536,18 +537,40 @@ local function stream_begin(stream, txn_opts, netbox_opts)
             txn_isolation =
                 box.internal.normalize_txn_isolation_level(txn_isolation)
         end
+        is_sync = txn_opts.is_sync
+        if type(is_sync) ~= "boolean" and is_sync ~= nil then
+            error("is_sync must be a boolean")
+        end
+        if is_sync == false then
+            error("is_sync can only be true")
+        end
     end
     local res = stream:_request('BEGIN', netbox_opts, nil,
-                                stream._stream_id, timeout, txn_isolation)
+                                stream._stream_id, timeout, txn_isolation,
+                                is_sync)
     if netbox_opts and netbox_opts.is_async then
         return res
     end
 end
 
-local function stream_commit(stream, opts)
+local function stream_commit(stream, txn_opts, opts)
```

Comment:
This changes the public user interface.

For example, [here](https://www.tarantool.io/en/doc/latest/reference/reference_lua/net_box/#overview) it's stated, that `Most net.box methods accept the last {options} argument`, and this is true for `stream:commit`.
But [stream.commit() doc](https://www.tarantool.io/en/doc/latest/reference/reference_lua/net_box/#lua-function.stream.commit) says `stream:commit()` has no arguments. So the user expects the first argument to be `{opts}`. But this is not true after your patch.

We need to document this change somehow and better make the change backward compatible. Users might already be using streams, we can't break them.

For example, you might try to understand if the second option is `opts` or `txn_opts`:
if the second option contains `is_sync` - it is txn_opts, it can't contain `is_async`,  and `opts` should be found in the 3rd argument.
If the second option contains `is_async` -- it is opts, it can't contain `is_sync`, and there should be no 3rd argument. The user can't pass `txn_opts` then.

Reply:
Done.